tw-tasks-executor/src/main/java/com/transferwise/tasks/management/TasksManagementPortController.java

```diff
@@ -22,121 +39,198 @@
-            .setTaskVersionId(taskInError.getTaskVersionId())
-            .setType(taskInError.getType()).setSubType(taskInError.getSubType())
-    ).collect(Collectors.toList()));
+  public ResponseEntity<GetTasksInErrorResponse> getTasksInError(@RequestBody(required = false) GetTasksInErrorRequest request) {
+    return callWithAuthentication(() -> {
+      ITasksManagementService.GetTasksInErrorResponse serviceResponse = tasksManagementService
+          .getTasksInError(new ITasksManagementService.GetTasksInErrorRequest()
+              .setMaxCount(request != null ? request.getMaxCount() : DEFAULT_MAX_COUNT));
+
+      return ResponseEntity.ok(new GetTasksInErrorResponse().setTasksInError(serviceResponse.getTasksInError().stream().map(taskInError ->
+          new GetTasksInErrorResponse.TaskInError()
+              .setErrorTime(taskInError.getErrorTime())
+              .setTaskVersionId(taskInError.getTaskVersionId())
+              .setType(taskInError.getType()).setSubType(taskInError.getSubType())
+      ).collect(Collectors.toList())));
+    });
   }
 
   @Override
-  public TaskWithoutData getTask(@PathVariable final String taskId) {
-    FullTaskRecord task = taskDao.getTask(UUID.fromString(taskId), FullTaskRecord.class);
-    return new TaskWithoutData()
-        .setId(taskId)
-        .setVersion(task.getVersion())
-        .setStatus(task.getStatus())
-        .setType(task.getType())
-        .setSubType(task.getSubType())
-        .setNextEventTime(Date.from(task.getNextEventTime().toInstant()))
-        .setStateTime(Date.from(task.getStateTime().toInstant()))
-        .setPriority(task.getPriority())
-        .setProcessingTriesCount(task.getProcessingTriesCount())
-        .setProcessingClientId(task.getProcessingClientId());
+  public ResponseEntity<TaskWithoutData> getTask(@PathVariable final String taskId) {
+    return callWithAuthentication(() -> {
+      FullTaskRecord task = taskDao.getTask(UUID.fromString(taskId), FullTaskRecord.class);
+      return ResponseEntity.ok(new TaskWithoutData()
+          .setId(taskId)
+          .setVersion(task.getVersion())
+          .setStatus(task.getStatus())
+          .setType(task.getType())
+          .setSubType(task.getSubType())
+          .setNextEventTime(Date.from(task.getNextEventTime().toInstant()))
+          .setStateTime(Date.from(task.getStateTime().toInstant()))
+          .setPriority(task.getPriority())
+          .setProcessingTriesCount(task.getProcessingTriesCount())
+          .setProcessingClientId(task.getProcessingClientId()));
+    });
   }
 
   @Override
-  public GetTasksInProcessingOrWaitingResponse getTasksInProcessingOrWaiting(
+  public ResponseEntity<GetTaskDataResponse> getTaskData(UUID taskId) {
+    return callWithAuthentication(tasksProperties.getTasksManagement().getViewTaskDataRoles(),
+        (auth) -> log.info("User was lacking roles '{}' trying to fetch payload for task '{}'.", auth.getName(), taskId),
```

Comment:
to me, it looks generic and could be in `callWithAuthentication`

Reply:
ok, removed the logs.